docs/adrs/0076-partial-update-patches.md

```diff
@@ -0,0 +1,146 @@
+	CounselorRemarks                 Nullable[*string]
+	PickupAddress                    *Address
+	PickupAddressID                  *uuid.UUID
+	DestinationAddress               *Address
+	DestinationAddressID             *uuid.UUID
+	DestinationType                  *DestinationType
+	MTOAgents                        MTOAgents
+	MTOServiceItems                  MTOServiceItems
+	SecondaryPickupAddress           *Address
+	SecondaryPickupAddressID         *uuid.UUID
+	SecondaryDeliveryAddress         *Address
+	SecondaryDeliveryAddressID       *uuid.UUID
+	SITDaysAllowance                 *int
+	SITExtensions                    SITExtensions
+	PrimeEstimatedWeight             *unit.Pound
+	PrimeEstimatedWeightRecordedDate *time.Time
+	PrimeActualWeight                *unit.Pound
+	BillableWeightCap                *unit.Pound
+	BillableWeightJustification      *string
+	NTSRecordedWeight                *unit.Pound
+	ShipmentType                     MTOShipmentType
+	Status                           MTOShipmentStatus
+	Diversion                        bool
+	RejectionReason                  *string
+	Distance                         *unit.Miles
+	Reweigh                          *Reweigh
+	UsesExternalVendor               bool
+	StorageFacility                  *StorageFacility
+	StorageFacilityID                *uuid.UUID
+	ServiceOrderNumber               *string
+	TACType                          *LOAType
+	SACType                          *LOAType
+	PPMShipment                      *PPMShipment
+	CreatedAt                        time.Time
+	UpdatedAt                        time.Time
+	DeletedAt                        *time.Time
+}
+```
+
+Handlers would be in charge of converting payloads into the MTOShipmentUpdate model and passing it to the UpdateShipment service. The UpdateShipment service would be responsible for fetching the database model (MTOShipment) and updating it appropriately based on the `Present` fields in the MTOShipmentUpdate.
+
+Such a design would have many advantages in addition to solving the problem of partial model updates and setting fields to null. The UpdateMTOShipment struct provides clarity by broadcasting intent: it clearly exists to update a shipment and it tells the service exactly what it wants. It also helps to further disentangle the handler layer from the database and services; since the handler now uses the MTOShipmentUpdate model, it no longer needs to know about the database structure. This design will also allow for a thorough cleanup of a lot of the handler and service logic through removal of repeated null checking and shifting business logic responsibilities cleanly to the services.
+
+### Implementation
+
+Due to the scope of work required, implementating this pattern would need to be done slowly.
+
+#### Decisions to finalize
+* Decide on a naming convention for intermediary update models
+* Determine if all models should fall under this pattern or only update models.
```

Comment:
I like these section, but I think in asking the questions you may have already answered them. The suggestion below is my re-wording, but there's no need to accept the changes.

```suggestion
#### Decisions to socialize
* The naming convention for intermediary update models will be `IntermediaryUpdateModels`.
* We'll do the `update` models first, but we will need measure if other models should also be a part of this similar pattern.
```

The reason I re-worded this is because I believe that even with a decision being made here, we can revisit any changes that need to be made in a subsequent ADR.

Reply:
I completely agree @shimonacarvalho - I'd like to find a good naming convention that describes what these are and their intended use case. In researching design pattern opportunities, I also locally kicked around "Payload Models" and "API Models". Both assume that our current Models package implies "Data Models", however which can still feel confusing and/or misleading. I can see a use case for non-update payloads (i.e. I've worked in such a pattern in the past) but the scope of implementing a pattern like that would be enormous, which is why I've gone back and forth.

Reply:
These are great callouts @rogeruiz and @shimonacarvalho , thank you!
With respect to naming, the update structs are _structs_, but they are not specifically _database models_. I've changed the language of the ADR to reflect that.
As for the decisions to finalize, I've taken yall's advice and removed that section and answered those questions directly in the ADR.
The "Requirements and Naming" section should address all of the above.